pkg/oc/cli/admin/release/new.go

```diff
@@ -438,16 +470,37 @@ func (o *NewOptions) Run() error {
 
 		fmt.Fprintf(o.ErrOut, "info: Found %d images in release\n", len(is.Spec.Tags))
 
-	case len(o.FromImageStream) > 0:
+	case len(o.FromImageStream) > 0, len(o.FromImageStreamFile) > 0:
 		is = &imageapi.ImageStream{}
 		is.Annotations = map[string]string{}
 		if len(o.FromImageStream) > 0 && len(o.Namespace) > 0 {
 			is.Annotations[annotationReleaseFromImageStream] = fmt.Sprintf("%s/%s", o.Namespace, o.FromImageStream)
 		}
 
-		inputIS, err := o.ImageClient.ImageV1().ImageStreams(o.Namespace).Get(o.FromImageStream, metav1.GetOptions{})
-		if err != nil {
-			return err
+		var inputIS *imageapi.ImageStream
+		if len(o.FromImageStreamFile) > 0 {
+			data, err := ioutil.ReadFile(o.FromImageStreamFile)
+			if os.IsNotExist(err) {
+				return err
+			}
+			if err != nil {
+				return fmt.Errorf("unable to read input image stream file: %v", err)
+			}
+			is := &imageapi.ImageStream{}
+			if err := yaml.Unmarshal(data, &is); err != nil {
+				return fmt.Errorf("unable to load input image stream file: %v", err)
+			}
+			if is.Kind != "ImageStream" || is.APIVersion != "image.openshift.io/v1" {
+				return fmt.Errorf("unrecognized input image stream file, must be an ImageStream in image.openshift.io/v1")
+			}
+			inputIS = is
+
+		} else {
```

Comment:
Instead of doing else withing case, why not making it a separate case? Also this means I can't have both, but I don't see any validation for that which means it will confuse users.

Reply:
The validation is "sources" up above.